packages/cli/src/reporters/json.ts

```diff
@@ -0,0 +1,95 @@
+import * as fs from 'fs'
+import * as path from 'path'
+
+import AbstractListReporter, { checkFilesMap } from './abstract-list'
+import { CheckRunId } from '../services/abstract-check-runner'
+import { formatDuration, printLn, getTestSessionUrl } from './util'
+
+const outputFile = './checkly-json-report.json'
+
+type JsonBuilderOptions = {
+  testSessionId?: string
+  numChecks: number
+  runLocation: string
+  checkFilesMap: checkFilesMap
+}
+
+export class JsonBuilder {
+  testSessionId?: string
+  numChecks: number
+  runLocation: string
+  checkFilesMap: checkFilesMap
+  hasFilenames: boolean
+
+  constructor (options: JsonBuilderOptions) {
+    this.testSessionId = options.testSessionId
+    this.numChecks = options.numChecks
+    this.runLocation = options.runLocation
+    this.checkFilesMap = options.checkFilesMap
+    this.hasFilenames = !(options.checkFilesMap.size === 1 && options.checkFilesMap.has(undefined))
+  }
+
+  render (): string {
+    const testSessionSummary: any = {
+      testSessionId: this.testSessionId,
+      numChecks: this.numChecks,
+      runLocation: this.runLocation,
+      checks: [],
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const [_, checkMap] of this.checkFilesMap.entries()) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      for (const [_, { result, testResultId }] of checkMap.entries()) {
+        const check: any = {
+          result: result.hasFailures ? 'Fail' : 'Pass',
+          name: result.name,
+          checkType: result.checkType,
+          duration: formatDuration(result.responseTime),
+          durationMilliseconds: result.responseTime,
```

Comment:
I think that we should just keep one of `duration` / `durationMilliseconds`

Reply:
Yeah, simpler is better. I agree.